### ISSUE TITLE
Make every concrete Feature a Mapping.

### ISSUE THREAD

**Issue by MatthieuDartiailh** [LabPy/lantz_core#7]
The idea behind that is that it is sometimes convenient to have a Float with a unit (for example for a range) but because the instrument uses only a discrete set of values it expects a meaningless value. Doing that would mean that a Float or Unicode can be a Mapping.

WDYT ?

**Comment by hgrecco** [comment]
But you will need to coerce the input value before going through the mapping, right?

**Comment by MatthieuDartiailh** [comment]
Yes of course this is what is currently happening when you specify an enumerated float, first the unit conversion is performed before checking that the value is allowed. So this is not an issue.

**Comment by hgrecco** [comment]
Would not this require to store a lot of values in a dictionary. For example if we accept floats from 0 to 10000 with 0.01 steps, might be a lot of values.

**Comment by MatthieuDartiailh** [comment]
My idea was to reserve this to cases where there are only few values which needs to be mapped to instrument values (ex : three ranges 10mV, 100mV, 1V but we need to send to the instrument 1, 2 or 3). If the mapping keyword is absent nothing happen. Another solution is to forbid it and say that the range is not a float but a string. I am not sure what is the best solution.

**Comment by hgrecco** [comment]
In that case I am all in. That is definitely a mapping. I see the Float (or Float + Unit) only for ranges with certain optional precision.

**Comment by MatthieuDartiailh** [comment]
I am not sure we understand completely each other. I will try to clarify with some pseudo-cpde : 
```
class DCSource():

    voltage = Float('Volt?', 'volt {}', limits=(0, 1, 0,1), unit='V')

    voltage_range = Float('volt:rang?', 'volt:rang {}', unit='V', mapping={0.1: 1, 1.0: 2})
```
In such a case setting the voltage range can be done by passing a Quantity of 100mV. The only interest is that we can leverage the unit handling from Float. That's why I am not sure how much sense it makes. I will try to make some progress on the dc sources without it and we will see what makes the more sense.

**Comment by hgrecco** [comment]
I understand and agree. 

You are saying doing this:
```
Float('volt:rang?', 'volt:rang {}', unit='V', mapping={0.1: 1, 1.0: 2})
```
instead of this:
```
Mapping('volt:rang?', 'volt:rang {}', mapping={0.1 * V: 1, 1.0 *V : 2})
```

which does not seem to be a great thing from the length of the user code, but might be useful from the implementation.

**Comment by MatthieuDartiailh** [comment]
Ok I will then assume that we will include this change and see how this turns out in the dc sources and the next drivers I will implement.

**Comment by MatthieuDartiailh** [comment]
I made the changes in #10

**Comment by MatthieuDartiailh** [comment]
Closed by #10
